core/src/impl/Kokkos_Core.cpp

```diff
@@ -271,7 +271,7 @@ int Kokkos::Impl::get_ctest_gpu(int local_rank) {
     ss << "Error: local rank " << local_rank
        << " is outside the bounds of resource groups provided by CTest. Raised"
        << " by Kokkos::Impl::get_ctest_gpu().";
-    throw_runtime_exception(ss.str());
+    abort(ss.str().c_str());
```

Comment:
`get_ctest_gpu` in `core/src/impl`, seriously ?

Reply:
Shouldn't it be some helpers in `core/unit_test` or something like that ?

Reply:
I think you misunderstand what this does.
This is part of the GPU selection logic at initialization.

Reply:
I still think that this is very odd.

If someone needs to set a GPU for testing, depending on a `CTest` resource group, they should implement their own logic to map to `KOKKOS_VISIBLE_DEVICES` probably.

Or the logic of `Kokkos::Impl::get_ctest_gpu` could be make available (tweaked a bit to map to `KOKKOS_VISIBLE_DEVICES`) but definitely very awkward to find that in `core` (because it is related to tsting, and because it is bound to `CTest`).

Reply:
Not the place to argue about this